### PR TITLE
security: update FastAPI

### LINKED ISSUE #377: Update fastapi - Add support to version >= 0.100.0
### What do you want to share with us?

The FastAPI 0.100.0 support pydantic 2.0

https://github.com/tiangolo/fastapi/releases/tag/0.100.0

This version changes the OpenAPI support in the RSTUF API, requiring changes in the API documentation.

### References

- https://github.com/tiangolo/fastapi/releases/tag/0.100.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by kairoaraujo**
# Description
https://fastapi.tiangolo.com/release-notes/#security-fixes

# Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

Closes #377

### CODE REVIEW

**Review by MVrachev — APPROVED** [review]

**Comment by MVrachev** [comment]
Ping me when the failures are resolved, so I can approve it again.

File: repository_service_tuf_api/bootstrap.py [hunk 1]
```diff
@@ -42,55 +43,50 @@ class Settings(BaseModel):
...
+    )
+    data: PostData | None = None
+    message: str
 
 
 class GetData(BaseModel):
     bootstrap: bool
-    state: Optional[str]
-    id: Optional[str]
+    state: str | None = None
```
Comment (lukpueh) [inline repository_service_tuf_api/bootstrap.py:75]: Why change type hint here and elsewhere?

File: repository_service_tuf_api/common_models.py [hunk 2]
```diff
@@ -73,33 +88,32 @@ class TUFSignedRoles(BaseModel):
...
     def validate_unrecognized_fields(
         cls, values: Dict[str, Any]
     ) -> Dict[str, Any]:
         all_required_field_names = {
-            field.alias for field in cls.__fields__.values()
+            v.alias or f for f, v in cls.model_fields.items()
         }
 
-        for field_name in list(values):
+        for field_name in list(values.keys()):
```
Comment (lukpueh) [inline repository_service_tuf_api/common_models.py:116]: Is this change needed?

**Review by lukpueh — APPROVED** [review]
Yay for pydantic v2! :)

File: repository_service_tuf_api/bootstrap.py (hunk 1 above)
Comment (kairoaraujo) [reply repository_service_tuf_api/bootstrap.py:75]: I thought it could be good to follow the FastAPI d[ocumentation for Python 3.10+](https://fastapi.tiangolo.com/tutorial/body/#__tabbed_1_1) as we support Python >=3.10

File: repository_service_tuf_api/common_models.py (hunk 2 above)
Comment (kairoaraujo) [reply repository_service_tuf_api/common_models.py]: It is not needed, fixed.

**Review by lukpueh — APPROVED** [review]
